### PR TITLE
Add post library API as custom mutator and rename pre_save

### PR DESCRIPTION
**Pull request by rish9101**
Renamed pre_save to something better.
Used the POST library api as a custom mutator since now we can support multiple custom mutators

### CODE REVIEW

File: examples/post_library/post_library.so.c [hunk 1]
```diff
@@ -83,7 +83,7 @@ typedef struct post_state {
 
 } post_state_t;
 
-void *afl_postprocess_init(void *afl) {
+void *afl_postprocess_init(void *afl, unsigned int seed{
```
Comment (vanhauser-thc) [inline examples/post_library/post_library.so.c:86]: have you compile this change? it wont - closing bracket missing

File: include/afl-fuzz.h [hunk 2]
```diff
@@ -260,7 +260,7 @@ enum {
 
   /* 00 */ PY_FUNC_INIT,
   /* 01 */ PY_FUNC_FUZZ,
-  /* 02 */ PY_FUNC_PRE_SAVE,
+  /* 02 */ PY_FUNC_post_process,
```
Comment (vanhauser-thc) [inline include/afl-fuzz.h:263]: capital letters please

File: src/afl-fuzz-init.c [hunk 3]
```diff
@@ -311,16 +315,18 @@ void setup_post(afl_state_t *afl) {
   /* Do a quick test. It's better to segfault now than later =) */
 
   u8 *post_buf = NULL;
-  afl->post_data = afl->post_init(afl);
-  if (!afl->post_data) { FATAL("Could not initialize post handler."); }
+  mutator->data = mutator->afl_custom_init(afl, rand_below(afl, 0xFFFFFFFF));
+  if (!mutator->data) { FATAL("Could not initialize post handler."); }
 
-  size_t post_len = afl->post_handler(afl->post_data, tbuf, tlen, &post_buf);
+  size_t post_len = mutator->afl_custom_post_process(mutator->data, tbuf, tlen, &post_buf);
```
Comment (vanhauser-thc) [inline src/afl-fuzz-init.c:321]: maybe remove the testing? we dont do that for normal custom mutators either.

**Comment by vanhauser-thc** [comment]
and I think you have to update the python document for the pre_save -> post_process change

**Comment by vanhauser-thc** [comment]
looks good otherwise :)

**Comment by rish9101** [comment]
> and I think you have to update the python document for the pre_save -> post_process change

Docs have been updated I think.

File: include/afl-fuzz.h [hunk 4]
```diff
@@ -260,7 +260,7 @@ enum {
 
   /* 00 */ PY_FUNC_INIT,
   /* 01 */ PY_FUNC_FUZZ,
-  /* 02 */ PY_FUNC_post_process,
+  /* 02 */ PY_FUNC_POST_PROCESS,
```
Comment (vanhauser-thc) [inline include/afl-fuzz.h:263]: You still have to update all locations where you have used the lowcase letter version.
Always, always compile before committing

**Comment by vanhauser-thc** [comment]
looks good now thanks.

**Pull request merged**